### DISCUSSION TITLE
Degrade PDE DS UI to a viewer of generated OSGi DS component XML files

### DISCUSSION THREAD

**Discussion by HannesWell** [eclipse-pde/eclipse.pde#1399] [Ideas]
At the moment it is possible to create OSGi service component definitions through a dedicated wizard and it is possible to edit existing component definitions through a corresponding editor when opening a component XML file (see the screenshots below).

But since it is possible for quite some time to use the much more convenient [OSGI Declarative Service Annotations](https://docs.osgi.org/specification/osgi.cmpn/8.0.0/service.component.html#d0e30885) directly within the Java classes implementing the components I see not much value in creating and editing DS components by hand instead of using the annotations.

Therefore I suggest to remove the `Component Definition` wizard entirely and to degrade the `Component Definition` file editor to a pure viewer. The `org.eclipse.pde.ds.ui` plugin would then only provide a read-only `Component Definition` viewer plus the outline contribution, similar to when like one would open the component XML file from a plugin in the TP.
Maybe the viewer will then also be made openable from the Manifest-Editor like proposed in https://github.com/eclipse-pde/eclipse.pde/issues/40, but that's an orthogonal issue.

This would reduce the maintenance effort and would simplify the UI for users a little bit.

What's your opinion on this? Any objections?

![grafik](https://github.com/user-attachments/assets/236b9a39-988b-4a33-8292-82b46796b91e)

![grafik](https://github.com/user-attachments/assets/4b683b3f-95c1-49bf-9e26-5fb6dec71a9a)

**Comment by vogella** [comment]
+1 for the proposal

**Comment by ruspl-afed** [comment]
This editor is very useful to understand available capabilities of component declaration, we need to consider that its removal will increase the   entry barrier.
However, I understand that all the good things require resources to support them in a functional shape, and we may have much more important things to focus on.

**Reply by HannesWell** [reply]
> This editor is very useful to understand available capabilities of component declaration, we need to consider that its removal will increase the entry barrier.

The proposal is to keep the current Editor as a Viewer of the generated component XML files. Just its capability to edit these files would then be removed. Inspecting the files or understanding them will still be possible.
